OpenRA.Mods.Common/Warheads/HealthPercentageDamageWarhead.cs

```diff
@@ -30,25 +30,38 @@ public override void DoImpact(WPos pos, Actor firedBy, IEnumerable<int> damageMo
 				world.WorldActor.Trait<WarheadDebugOverlay>().AddImpact(pos, Spread, DebugOverlayColor);
 
 			var range = Spread[0];
-			var hitActors = world.FindActorsInCircle(pos, range);
+			var hitActors = world.FindActorsOnCircle(pos, range);
 			if (Spread.Length > 1 && Spread[1].Length > 0)
 				hitActors = hitActors.Except(world.FindActorsInCircle(pos, Spread[1]));
 
 			foreach (var victim in hitActors)
-				DoImpact(victim, firedBy, damageModifiers);
+			{
+				// Cannot be damaged without a Health trait
+				if (!victim.Info.HasTraitInfo<HealthInfo>())
+					continue;
+
+				if (!IsValidAgainst(victim, firedBy))
+					continue;
+
+				var impactToVictimDistance = new WDist((victim.CenterPosition - pos).Length);
```

Comment:
You take the `.Length`, wrap it in a `WDist`, only to call `.Length` on that again below!
IMO use `(victim.CenterPosition - pos).Length` directly in the lerp.